packages/valory/skills/twitter_scoring_abci/behaviours.py

```diff
@@ -308,14 +310,43 @@ def _get_twitter_registrations(self) -> Generator[None, None, Dict]:
 
             break
 
-        wallet_to_ids = self._get_wallet_to_ids(registrations)
+        self.context.logger.info(
+            f"Got Twitter potential registrations: {registrations}"
+        )
+
+        wallet_to_ids = yield from self._get_wallet_to_ids(registrations)
 
         self.context.logger.info(f"Got Twitter registrations: {wallet_to_ids}")
 
         return {
             "wallet_to_users": wallet_to_ids,
         }
 
+    def check_whitepaper_link(self, text: str) -> Generator[None, None, bool]:
+        """Check whether a text contains a shortened t.co link that points to the whitepaper"""
+
+        # Check there is a t.co link
+        match = re.search(TCO_LINK_REGEX, text)
+        if not match:
+            return False
+
+        link = match.group().strip()
+
+        self.context.logger.info(f"Found url: {link}")
+
+        # Visit the link
+        response = yield from self.get_http_response(
+            method="GET",
+            url=link,
+        )
+
+        # Check response status
+        if response.status_code != 200:
+            self.context.logger.error(f"Could not visit url: {link} [{response}]")
+            return False
+
+        return "Whitepaper | Autonolas" in response.body.decode("utf-8")
```

Comment:
Checking for "Whitepaper | Autonolas" does not seem very robust :sweat_smile:

Reply:
Yes, I agree. The original implementation was to check the inbound url (I tested that with the requests library) but the inbound messages do not have the url set, so I need to check some content because only the body is available.